sympy/functions/elementary/exponential.py

```diff
@@ -529,8 +529,13 @@ def _taylor(self, x, n):
         return Add(*l)
 
     def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        from sympy.calculus.util import AccumBounds
         arg = self.args[0].cancel().as_leading_term(x, logx=logx)
         arg0 = arg.subs(x, 0)
+        if arg is S.NaN:
+            return S.NaN
+        if isinstance(arg0, AccumBounds) and arg0.min == S.NegativeInfinity:
```

Comment:
Okay I'll try to convey my understanding of this to you .

**Overview:** Consider the case `exp(x*log(cos(1/x) + n)).as_leading_term(x)` . We always calculate leading term of `self.args[0]` first(stored in `arg`). Then we substitute `x` as `0` in arg(stored in `arg0`)

**Case1:** **exp(x*log(cos(1/x))).as_leading_term(x)**

- arg = x * log(cos(1/x)).as_leading_term(x) = nan 
- arg0 = arg.subs(x, 0) = nan
- we return nan from the first if statement (if arg == S.NaN: return S.NaN)

**Case2:** **exp(x*log(cos(1/x) + 2)).as_leading_term(x)**

- arg = x * log(cos(1/x) + 2).as_leading_term(x) = x*AccumBounds(0, log(3)) 
- arg0 = arg.subs(x, 0) = 0*AccumBounds(0, log(3)) = 0
- we return exp(0) from the fourth if statement (if arg0.is_infinite is False: return exp(arg0)

**Case2:** **exp(x*log(cos(1/x) + 1)).as_leading_term(x)** (Corner Case)

- arg = x * log(cos(1/x) + 1).as_leading_term(x) = x*AccumBounds(-oo, log(2))
This is because we return AccumBounds(-oo, log(x)) for log(AccumBounds(0, x)).

- arg0 = arg.subs(x, 0) = 0*AccumBounds(-oo, log(2)) = AccumBounds(-oo, 0)
- we return the answer from the new if statement implemented base on AccumBounds !

Reply:
Hence your doubt is completely valid . We will be encountering `0* AccumBounds` for `arg0` and this will only end up not being `0` when `AccumBounds.min == -oo` . Otherwise be it AccumBounds or anything multiplied by `0` , we would end up `0` unless it would be something like `zoo, oo , nan` . So maybe we just need to check whether `arg0` is AccumBounds , cause for all other cases like the 2nd case we will end up with `arg0` being `0` .
Ps: I tried my best to explain haha ! Hope it helps

Reply:
> I don't think there would be any case where arg0.min is not -oo 

So it seems that it would not be necessary to test `arg0.min == -oo`, `isinstance(arg0, AccumBounds)` alone should suffice.

Reply:
True agreed , had just put it for safety !but I realize that being a corner case it's not required .

Reply:
Both the new tests added are also corner cases (will periodically fall to 0 steeply in pi or 2*pi period)
I'll explain one of those here 
```
>>> limit((tan(x)**2)**(2/x) , x, oo)
AccumBounds(0, oo)
```
**Step 1)** At the final step we would end up calculating `log(tan(1/x)**2).as_leading_term(x)` where we would calculate `(tan(1/x)**2).as_leading_term(x)` and as the result i.e the `arg` we get is of type `AccumBounds` , we return ` log(arg)`
```
>>> (tan(1/x)**2).as_leading_term(x)
AccumBounds(0, oo)
>>> log(tan(1/x)**2).as_leading_term(x)
AccumBounds(-oo, oo)
```
**Step 2)** At the second last step this would involve calculating the leading term of `exp(2*x*log(tan(1/x)**2))` where the `arg` returned from the above step is `2*x*AccumBounds(-oo, oo)` . Now `arg0 ` would be `0*AccumBounds(-oo, oo)` which is ` AccumBounds(-oo, oo)` and this also is solved through the corner case implementation where `arg0` is on type `AccumBounds`

**Note :The corner case would only involve 2 cases one where we have `AccumBounds(-oo, 0)` and other where we have `AccumBounds(-oo, oo)`, hence we can always return `AccumBounds(0, exp(arg0.max))` .** 
And hence in this case we get `AccumBounds(0, oo)` as our answer for limit .

Reply:
And as expected we have our graph for `(tan(x)**2)**(2/x)` as follows . This rise steeply with infinite slope at every `n*pi/2`  and goes to zero steeply at every integer `pi`
![image](https://user-images.githubusercontent.com/87052487/154798287-b5070b25-8656-43fa-a634-fc98be329027.png)

The other new test case added (also a corner case)
```
>>> limit((sin(x)**2)**(1/x) , x, oo)
AccumBounds(0, 1)
```
![image](https://user-images.githubusercontent.com/87052487/154798328-3c289614-2a4c-405f-922a-c60bb614e500.png)

Reply:
I am quite happy with the latest commit but would like to hear from your side too . The corner case not being very intuitive I have added a set of comments which represent the changes made and the reason behind them, so that it's easy for Us to address this case in future if we make changes in `AccumBounds` etc
I am happy with the latest commit But would like to hear from your side ! @jksuom